File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.ts
function evalRPN(tokens: string[]): number {
    const set = new Set(['+', '-', '*', '/'])
    let stack = []
    let ans = 0

    for (const token of tokens) {
        if (!set.has(token)) {
            stack.push(Number(token))
        } else {
            let b = stack.pop()
            let a = stack.pop()
            
            if (token == "+") stack.push(a+b)
            if (token == "-") stack.push(a-b)
            if (token == "*") stack.push(a*b)
            if (token == "/") stack.push(Math.trunc(a/b))
        }
    }

    return stack[0]
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        s = []
        for token in tokens:
            try: 
                int(token)
                s.append(int(token))
            except ValueError:
                val1 = s.pop()
                val2 = s.pop()
                if token == '+':
                    curr = val2 + val1
                elif token == '-':
                    curr = val2 - val1
                elif token == '*':
                    curr = val2 * val1
                else:
                    curr = int(val2 / val1)
                s.append(curr)

        return s[0]